framework/include/base/MooseObject.h

```diff
@@ -55,297 +48,28 @@ class MooseObject : public ConsoleStreamInterface,
-  }
-
-  template <typename... Args>
-  void mooseInfo(Args &&... args) const
-  {
-    moose::internal::mooseInfoStream(_console, std::forward<Args>(args)...);
-  }
-
-  /**
-   * A descriptive prefix for errors for this object:
-   *
-   * The following <error_type> occurred in the object "<name>", of type "<type>".
-   */
-  std::string errorPrefix(const std::string & error_type) const;
+  using MooseBaseErrorInterface::_console;
+  using MooseBaseErrorInterface::mooseDeprecated;
+  using MooseBaseErrorInterface::mooseError;
+  using MooseBaseErrorInterface::mooseWarning;
+  using MooseBaseParameterInterface::getCheckedPointerParam;
+  using MooseBaseParameterInterface::getParam;
+  using MooseBaseParameterInterface::paramError;
+  using MooseBaseParameterInterface::paramInfo;
+  using MooseBaseParameterInterface::paramWarning;
 
 protected:
-  /// Parameters of this object, references the InputParameters stored in the InputParametersWarehouse
-  const InputParameters & _pars;
-
-  /// The MooseApp this object is associated with
-  MooseApp & _app;
-
-  /// The type of this object (the Class name)
-  const std::string & _type;
-
-  /// The name of this object, reference to value stored in InputParameters
-  const std::string & _name;
-
-  /// Reference to the "enable" InputParaemters, used by Controls for toggling on/off MooseObjects
+  /// Reference to the "enable" InputParameters, used by Controls for toggling on/off MooseObjects
   const bool & _enabled;
 
-private:
-  template <typename... Args>
-  std::string paramErrorMsg(const std::string & param, Args... args) const
-  {
-    auto param_prefix = paramErrorPrefix(_pars, param);
-
-    // With no input location information, append object info (name + type)
-    const std::string object_prefix =
-        _pars.inputLocation(param).empty() ? errorPrefix("parameter error") : "";
-
-    std::ostringstream oss;
-    moose::internal::mooseStreamAll(oss, std::forward<Args>(args)...);
-    std::string msg = oss.str();
-
-    // Wrap error message to a separate line from param_prefix if it is about to
-    // blow past 100 chars.  But only wrap if the param_prefix is long enough (12
-    // chars) for the wrap to buy us much extra length. We don't check object_prefix
-    // here because it will go before the parameter error
-    if ((param_prefix.size() > 12 && msg.size() + param_prefix.size() > 99) ||
-        msg.find("\n") != std::string::npos)
-    {
-      if (param_prefix.size() > 0 && param_prefix[param_prefix.size() - 1] != ':')
-        param_prefix += ":";
-      return object_prefix + param_prefix + "\n    " + MooseUtils::replaceAll(msg, "\n", "\n    ");
-    }
-    return object_prefix + param_prefix + " " + msg;
-  }
+  using MooseBase::_app;
+  using MooseBase::_name;
+  using MooseBase::_type;
+  using MooseBaseParameterInterface::_action_factory;
+  using MooseBaseParameterInterface::_factory;
```

Comment:
and here. There's no template parameter dependent base, so this shouldn't be necessary from that regard